sdks/python/build.gradle

```diff
@@ -268,32 +269,37 @@ task directRunnerIT(dependsOn: 'installGcpTest') {
 //
 //    ./gradlew :beam-sdks-python:portableWordCount -PjobEndpoint=localhost:8099
 //
-task portableWordCount(dependsOn: 'installGcpTest') {
-  doLast {
-    // TODO: Figure out GCS credentials and use real GCS input and output.
-    def options = [
-            "--input=/etc/profile",
-            "--output=/tmp/py-wordcount-direct",
-            "--runner=PortableRunner",
-            "--experiments=worker_threads=100",
-    ]
-    if (project.hasProperty("streaming"))
-      options += ["--streaming"]
-    else
+task portableWordCount {
+  dependsOn portableWordCountTask('portableWordCountExample', project.hasProperty("streaming"), envdir)
+}
+
+def portableWordCountTask(name, streaming, envdir) {
+  tasks.create(name) {
+    dependsOn = ['installGcpTest']
+    mustRunAfter = [':beam-runners-flink_2.11-job-server-container:docker', ':beam-sdks-python-container:docker']
+    doLast {
+      // TODO: Figure out GCS credentials and use real GCS input and output.
+      def options = [
+              "--input=/etc/profile",
+              "--output=/tmp/py-wordcount-direct",
+              "--runner=PortableRunner",
+              "--experiments=worker_threads=100",
+      ]
+      if (streaming)
+        options += ["--streaming"]
+      else
       // workaround for local file output in docker container
-      options += ["--environment_cache_millis=10000"]
-    if (project.hasProperty("jobEndpoint"))
-      options += ["--job_endpoint=${project.property('jobEndpoint')}"]
-    exec {
-      executable 'sh'
-      args '-c', ". ${envdir}/bin/activate && python -m apache_beam.examples.wordcount ${options.join(' ')}"
-      // TODO: Check that the output file is generated and runs.
+        options += ["--environment_cache_millis=10000"]
+      if (project.hasProperty("jobEndpoint"))
+        options += ["--job_endpoint=${project.property('jobEndpoint')}"]
+      exec {
+        executable 'sh'
+        args '-c', ". ${envdir}/bin/activate && python -m apache_beam.examples.wordcount ${options.join(' ')}"
```

Comment:
This pipeline does not have unbounded sources - does it actually make sense to run it in streaming mode on portable runner? I think we would want to run something like https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/streaming_wordcount.py for streaming usecase.

Reply:
cc: @angoenka

Reply:
I think it does make sense because batch and streaming are two different execution modes, both in Flink as well as in the Flink Runner. It doesn't matter much that the data is bounded, as all execution code can be executed. That is not to say the tests cannot be more elaborate.